File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        res = False
        appearingtime = {

        }
        for i in s:
            if i not in appearingtime:
                appearingtime[i] = 1
            else:
                appearingtime[i] += 1
        for i in t:
            if (i not in appearingtime) or appearingtime[i] == 0:
                return False
            else:
                appearingtime[i] -= 1
        for i in appearingtime:
            if appearingtime[i]!=0:
                return False
        return True

File: Data Structures & Algorithms/decode-ways/submission-3.cpp
class Solution {
public:
    int numDecodings(string s) {
        int dp[100];

        dp[0]=1;
        
        if(s[0]=='0'){
            dp[0]=0;
            dp[1]=0;
        }
        else if(s[1]=='0'){
            
                dp[0]=1;
                dp[1]=0;
            if(s[0]=='1' or s[0]=='2'){
                dp[1]=1;
                }
        }
        else if(s[0]=='1' or (s[0]=='2' and s[1]<='6')){
            dp[1]=2;
        }else{
            dp[1]=1;
        }
        
        for(int i=2;i<s.size();i++){
            if(s[i]=='0' ){
                if(s[i-1]=='1' or (s[i-1]=='2' and s[i]<='6')){
                dp[i]=dp[i-2];}
                else{
                    dp[i]=0;
                }
            }
            else if(s[i-1]=='1' or (s[i-1]=='2' and s[i]<='6')){
                dp[i]=dp[i-1]+dp[i-2];
            }
            else{
                dp[i]=dp[i-1];
            }
            
        }
        return dp[s.size()-1];
    }
};
